pkg/rpc/client/wsclient.go

```diff
@@ -113,6 +116,9 @@ func NewWS(ctx context.Context, endpoint string, opts Options) (*WSClient, error
 // Close closes connection to the remote side rendering this client instance
 // unusable.
 func (c *WSClient) Close() {
+	if !c.closeCalled.CAS(false, true) {
```

Comment:
The problem is that the code calling `Close` would think that `WSClient` is closed by the time it returns from `Close()`, while in fact it could still be active (another thread waiting on `c.done`. Maybe we better wait for it anyway.